redis/commands/core.py

```diff
@@ -4663,7 +4663,7 @@ def __init__(self, registered_client, script):
         if isinstance(script, str):
             # We need the encoding from the client in order to generate an
             # accurate byte representation of the script
-            encoder = registered_client.connection_pool.get_encoder()
+            encoder = registered_client.get_encoder()
```

Comment:
This is technically a breaking change. If any consumers of `redis-py` use Scripts directly, injecting their own `registered_client` that _isn't_ a subclass of `Redis` or `RedisCluster`, this could break for them if they haven't got `client.get_encoder()` defined.

Reply:
While I believe it's very unlikely - I'd rather not break the universe. Perhaps we could wrap this with an _if getattr_, check. What do you think @jakebarnwell

Reply:
That's a good idea! I will try something like that